### ISSUE TITLE
How to theme switch when using FormBuildSwitch (another package)

### ISSUE THREAD

**Issue by Alan386** [rydmike/flex_color_scheme#12]
Hello, I'm converting over to the color scheme package and everything is working out well except for switches. I use Flutter Form Builder a lot in my app. The switches are not recognizing the color scheme. What is the proper way to use the package with a standard switch (or another package switch)?

**Comment by rydmike** [comment]
Hi @Alan386, thanks for your question.

With form builder I assume you mean this package?
https://pub.dev/packages/flutter_form_builder

You can certainly use it or any kind of switch to toggle your Flutter application theme. It sounds to me like you are perhaps struggling a bit with state management in your application?

To make your application toggle themes dynamically you need to tie the switch values that control your user's input changes, to some state variable(s) that are used to toggle values in your MaterialApp's `theme` and `darkTheme` properties. This could then be for several different properties in FlexColorScheme.

In the five examples bundled with the package, this is done by passing down the state and using callbacks. This is of course not so nice in a larger application, or if you control a lot of properties. If you take a look at example 5 you can see why it becomes a nightmare to maintain something like it very quickly. 

You can certainly use example 5 to give you an idea on how to set something up where you can use all built in themes plus a few custom ones and pretty much all other available properties in FlexColorScheme to dynamically allow user to change them all at run time. To make it nice and cleaner, you should of course use some state management solution, instead of passing down state like the examples do.

If you are not using any state management solution in your app. I recommend that you start by looking into Provider to manage state in you Flutter app. I prefer using Riverpod, it is a bit better and more powerful imo. There are many other state management solutions for Flutter that work just as well too if you prefer them.

In the next release of FlexColorScheme I also plan to add a new example to demonstrate how I do it in my applications with Riverpod. However, explaining all the details here, is a bit beyond the scope of FlexColorScheme and not really related to using FlexColorScheme as such, it is more of a general Flutter programming and architecture topic.

**Comment by Alan386** [comment]
Hi @rydmike,

I'm using Get for state management. I previously used Provider and RiverPod. The this app Get was the best fit. I'm able to switch between dark and light modes with no problem. All the controls change except for the switches which stay the same color in dark and light modes.

The issue I'm having is I set the FlexColor scheme to bigStone (both light and dark). I then added a simple switch widget like this:

                  Switch(value: true, onChanged: (v) {}),

I made sure I don't have any styling going on. The color of the switch is orange which is pretty close to Big Stone but when changed the default to Sakura it didn't change. It's still orange. I completed restarted the app.

I'm sure I'm not doing something correctly. I'll keep testing.
